app/src/main/java/org/p2p/wallet/qr/ui/ScanQrFragment.kt

```diff
@@ -52,10 +72,26 @@ class ScanQrFragment :
 
     private val barcodeCallback: ZXingScannerView.ResultHandler =
         ZXingScannerView.ResultHandler { rawResult ->
-            rawResult?.text?.let {
-                successCallback?.invoke(it)
-                popBackStack()
-            }
+            val continueAction: () -> Unit = { startCameraPreview() }
+            rawResult?.text?.let { address ->
+                try {
+                    PublicKey(address)
+                    setFragmentResult(requestKey, bundleOf(resultKey to address))
+                    popBackStack()
+                } catch (e: Throwable) {
+                    showUiKitSnackBar(
+                        messageResId = R.string.qr_no_address,
+                        actionButtonResId = android.R.string.ok,
+                        onDismissed = continueAction,
+                        actionBlock = { continueAction.invoke() }
+                    )
+                }
+            } ?: showUiKitSnackBar(
+                messageResId = R.string.qr_common_error,
+                actionButtonResId = android.R.string.ok,
+                onDismissed = continueAction,
+                actionBlock = { continueAction.invoke() }
+            )
```

Comment:
hard to read. 
maybe you should check `rawResult` for null first?

Reply:
hmm ok i will extract it to methods